addons/purchase/controllers/portal.py

```diff
@@ -111,6 +110,34 @@ def portal_my_purchase_order(self, order_id=None, access_token=None, **kw):
 
         report_type = kw.get('report_type')
         if report_type in ('html', 'pdf', 'text'):
-            return self._show_report(model=order_sudo, report_type=report_type, report_ref='purchase.action_report_purchase_order',download=kw.get('download'))
+            return self._show_report(model=order_sudo, report_type=report_type, report_ref='purchase.action_report_purchase_order', download=kw.get('download'))
+
+        confirm_type = kw.get('confirm')
+        if confirm_type == 'reminder':
+            order_sudo.confirm_reminder_mail()
+        if confirm_type == 'purchase_order':
+            order_sudo.confirm_purchase_order_mail()
+
         values = self._purchase_order_get_page_view_values(order_sudo, access_token, **kw)
-        return request.render("purchase.portal_my_purchase_order", values)
+        update_date = kw.get('update')
+        if update_date == 'True':
```

Comment:
'True' is a fishy value to check, however i'm not familiar with this kind of front-end thing
kw isn't json.loaded ? do other front-end controller maybe call json.loads on kw ?

Reply:
kw is a dict not a json obj, but yeah, I also think it's not good.